src/model/image-tag.ts

```diff
@@ -100,8 +109,12 @@ class ImageTag {
         return windows;
       case Platform.types.StandaloneLinux64: {
         // Unity versions before 2019.3 do not support il2cpp
-        if (process.env.USE_IL2CPP === 'true' && (major >= 2020 || (major === 2019 && minor >= 3))) {
-          return linuxIl2cpp;
+        if (major >= 2020 || (major === 2019 && minor >= 3)) {
+          if (providerStrategy === 'local') {
+            return linuxIl2cpp;
+          } else {
+            return process.env.USE_IL2CPP === 'true' ? linuxIl2cpp : linux;
+          }
         }
```

Comment:
nitpick: I asked myself if this would be easier to read this way:

```suggestion
        const isVersionEligible = major > 2019 || (major === 2019 && minor >= 3);
        const shouldUseIl2cpp = providerStrategy === 'local' || process.env.USE_IL2CPP === 'true';
        
        if (isVersionEligible) {
          return shouldUseIl2cpp ? linuxIl2cpp : linux;
        } else {
          return linux;
        }
```

but after comparing this suggestion, I think tu current code is still better and in line with other cases from that switch case 👍  Looks good to me.